GitUI/CommandsDialogs/FormCleanupRepository.cs

```diff
@@ -164,11 +164,29 @@ private void AddExcludePath_Click(object sender, EventArgs e)
 
             if (path is not null)
             {
-                path = path.Replace(Module.WorkingDir, "");
+                path = MakePathRelativeToModule(path);
                 textBoxExcludePaths.Text += path;
             }
         }
 
+        private string MakePathRelativeToModule(string targetPath)
+        {
+            targetPath = targetPath.Replace(Module.WorkingDir, "").ToPosixPath();
+            List<string> submodulePaths = (List<string>)Module.GetSubmodulesLocalPaths();
+
+            string nearestSubmodule = submodulePaths
+                .Where(submodulePath => targetPath.StartsWith(submodulePath))
+                .OrderByDescending(submodulePath => submodulePath.Length)
+                .FirstOrDefault();
+
+            if (nearestSubmodule is null)
+            {
+                return targetPath;
+            }
+
+            return targetPath.Substring(nearestSubmodule.Length + 1);
```

Comment:
We have switched to the new syntax:
```suggestion
            return targetPath[(nearestSubmodule.Length + 1)..];
```